server/base/src/main/java/org/apache/accumulo/server/compaction/CompactionJobGenerator.java

```diff
@@ -205,9 +209,22 @@ private Collection<CompactionJob> planCompactions(CompactionServiceId serviceId,
         // remove any files that are in active compactions
         tablet.getExternalCompactions().values().stream().flatMap(ecm -> ecm.getJobFiles().stream())
             .forEach(tmpFiles::remove);
-        // remove any files that are selected
-        if (tablet.getSelectedFiles() != null) {
-          tmpFiles.keySet().removeAll(tablet.getSelectedFiles().getFiles());
+        // remove any files that are selected and the user compaction has completed
+        // at least 1 job, otherwise we can keep the files
+        var selectedFiles = tablet.getSelectedFiles();
+
+        long selectedExpirationDuration =
```

Comment:
Could move this into the `if (selectedFiles != null) {` to avoid looking up this config when its not needed.